text/0000-deprecate-array-prototype-extensions.md

```diff
@@ -0,0 +1,130 @@
+---
+Stage: Initial
+Start Date: 2022-4-01
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js
+RFC PR: 
+---
+
+# Deprecate array prototype extensions
+
+## Summary
+
+This RFC proposes to deprecate array prototype extensions.
+
+## Motivation
+
+Ember historically extended the prototypes of native Javascript arrays to implement `Ember.Enumerable`, `Ember.MutableEnumerable`, `Ember.MutableArray`, `Ember.Array`. This added convenient methods and properties, and also made Ember arrays automatically participate in the Ember Classic reactivity system.
+
+Those convenient methods increase the likelihood of becoming potential roadblocks for future built-in language extensions, and make it confusing for users to onboard: is it specifically part of Ember, or Javascript? With Ember Octane, the new reactivity system, those classic observable-based methods are no longer needed.
+
+We had deprecated [Functions](https://github.com/emberjs/rfcs/blob/master/text/0272-deprecation-native-function-prototype-extensions.md) and [Strings](https://github.com/emberjs/rfcs/blob/master/text/0236-deprecation-ember-string.md) prototype extensions. Array is the last step. And internally we had already been preferring generic array methods over prototype extensions ([epic](https://github.com/emberjs/ember.js/issues/15501)).
+
+Continuing in that direction, we should consider recommending the usage of generic array functions as opposed to convenient prototype extension functions, and the usage of new tracked properties over classic reactivity methods.
```

Comment:
In the guide https://guides.emberjs.com/release/configuring-ember/disabling-prototype-extensions/#toc_arrays, Ember A() is still an option. Shall we recommending not using ember A() and deprecating it as well?

Reply:
I missed this, but: yes, absolutely, we should do so. One emberjs/rfcs#812 lands, there is *really* no need for it.

Reply:
Got it. Sounds like the discussion of `A()` deprecation can be in a separate RFC. This RFC focuses on deprecation of the default array prop extensions for ember app.